openssl/src/x509/extension.rs

```diff
@@ -506,14 +507,23 @@ impl SubjectAlternativeName {
 
     /// Sets the `otherName` flag.
     ///
-    /// Not currently actually supported, always panics.
-    #[deprecated = "other_name is deprecated and always panics. Please file a bug if you have a use case for this."]
+    /// Not currently actually supported, always panics. Please use other_name2
+    #[deprecated = "other_name is deprecated and always panics. Please use other_name2."]
     pub fn other_name(&mut self, _other_name: &str) -> &mut SubjectAlternativeName {
         unimplemented!(
             "This has not yet been adapted for the new internals. File a bug if you need this."
         );
     }
 
+    /// Sets the `otherName` flag to a ia5string value.
+    pub fn other_name_string(&mut self, oid: &str, content: &str) -> &mut SubjectAlternativeName {
```

Comment:
I think we want to use more precise types here: oid should be an Asn1ObjectIdentifier type, and contents should be `&[u8]` since it's DER.

Reply:
IMO limiting to strings doesn't very much sense, OtherName is a very general purpose specification, and we should have an API that's similarly broad. Why do you think `unsafe` would be required?

Reply:
thats why i would allow both. To get the combination of ease of use and all the features you might need.

I see `unsafe` as required as the data would then be passed to `GENERAL_NAME_set0_othername`. In there the openssl lib expects (if i understood it correctly) certain data formats (e.g. during deserialization). At least the last parameter must be a `ASN1_TYPE` as otherwise you will get a SegmentationFault (which is a clear sign to me that we need unsafe here, or guarantee that we do not need it like in the string case)

Reply:
If the public API takes `&[u8]`, a sequence of DER bytes, then it will be the responsibility of rust-openssl to have the unsafe code to turn that into an `ASN1_TYPE`, we can do that with `d2i_ASN1_TYPE`.

Reply:
Thanks for the clarification.

I have now added `other_name2` that will take a provided `&[u8]` and try to convert it to an `ASN1_TYPE`.

For ease of use i have adapted the `other_name_ia5string` method, but if you want to get rid of it that is also fine for me.